### ISSUE TITLE
Speech Bubble Detection - training issue

### ISSUE THREAD

**Issue by Little-BigMan** [Little-BigMan/YOLOv4#1]
> YOLOv4의 custom training issue 

# YOLOv4 말풍선 탐지 모델 학습 이슈

### 학습 옵션 조절 
+ cfg.py
    + class 1 (해당 모델은 말풍선만 탐지하는 모델)
    + learning_rate 0.001
    + max_batches = 2000  (class * 2000)
    + steps = [1600, 1800], (max_batches * 0.8 , max_batches * 0.9)

+ cfg/yolov4.cfg
    + class 1
    + filter 18 (4 + 1 + class) * 3  수정 라인: 961, 1049, 1137

+ train.py
    + YOLOv4의 경우 iteration마다 learning rate을 조절함. 
        + if i < burn_in(cfg.py) -> learning rate * (iteration/burn_in)^4
        + elif i < step[0](1600) -> learning rate * 1.0
        + elif i < step[1](1800) -> learning rate * 0.1
        + else learning rate * 0.01
   + 실제 YOLOv4 학습시간 때문에 learning rate * 1.0으로 통일  
 
## 2. training 

+ image size 608  O
+ image size 416  X 학습이 안되는 오류 존재
+  ```./train.sh``` 실행
    + ``` -dir /dataset_path/ -pretrained /pretarined_model_path/yolov4.conv.137.pth ``` 만 설정. 
    + 수정 : self.ROOT에 `-dir` 값 넣어줘야함 <br>     
    + `-dir`: 데이터가 있는 폴더 위치 해당 폴더안에는 image, XML 두가지 폴더가 있어여한다.(폴더명도 일치해야함)
    + `-pretrained` : yolov4.conv.137.pth 위치   

# demo

+ demo image size 320, 416, 512, 608
+ demo code : gitlab에 업데이트 되어있음.
